### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by Calvinoea**
Hello,

We are making a pull request for the Bare Metals and Forms Helper project, in accordance with the requirements of Microverse and Odin. The project was completed using Rails 6 and Ruby 2.6.6. 

The project required us to carry out the following tasks:

- Build HTML Form
- Build railsy Forms with #form_tag
- Build railsy-er Forms with #form_for
- Build form for Editing

Checks have been made and passed with Rubocop. 

Please find below a link to the ReadMe file with more details of the project:

-  [ReadMe](https://github.com/Calvinoea/forms-and-helpers/blob/feature/README.md)

Thank you for your time and review.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+    if @user.save
+      redirect_to new_users_path
```
Comment (stratospherique) [inline app/controllers/users_controller.rb]: - Invalid route :red_circle: It should be new_user_path

File: app/views/users/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+
+<%= form_for @user do |f| %>
+  <% if user.errors.any? %>
```
Comment (stratospherique) [inline app/views/users/edit.html.erb:3]: - @user not "user", this line is causing you app to crash. :red_circle:

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great start @team :clap: :clap: However, some requirements need to be implemented in order to get approval for this project. Check :arrow_down: 

#### Views
- Commented out forms are missing :warning: 
##### `new`

###### commented out `form`

* [x] uses `users_path` helper for action
* [x] uses `post` for method
* [x] uses `"UTF-8"` for ccept-charset
* [x] uses `user[<field_name>]` for input's name
* [x] uses form_authenticity_token

```html
<form action=users_path method='post' accept-charset="UTF-8">
  <label for='user[username]'>Username</label>
  <input name='user[username]' type='text'>
  <input type="hidden" name="authenticity_token" value="<%= form_authenticity_token %>">
  <input type="submit" value="Submit">
</form>
```

([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

###### commented out `form_tag`

* [x] uses `label_tag`, `text_field_tag`, `email_field_tag`, `password_field_tag`

```html
<%= form_tag(users_path) do |f|%>
  <%= label_tag 'username' %>
  <%= text_field_tag 'username' %>
  <%= label_tag 'email' %>
  <%= email_field_tag 'email' %>
  <%= label_tag 'password' %>
  <%= password_field_tag 'password' %>
  <%= submit_tag %>
<% end %>
```

([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_tag2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

###### `form_for`
- Errors get automatically generated in the view when using form_for
* [x] uses `@user` from controller

```html
<%= form_for @user do |u|%>
  <%= u.label :username %>
  <%= u.text_field :username %>
  <%= u.label :email %>
  <%= u.email_field :email %>
  <%= u.label :password %>
  <%= u.password_field :password %>
  <%= u.submit %>
<% end %>
```

([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_for.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

##### `edit`

([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_edit.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

### That's it :relieved: When finished, kindly submit a code re-review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Comment by Calvinoea** [comment]
> ### Great start @team 👏 👏 However, some requirements need to be implemented in order to get approval for this project. Check ⬇️
> #### Views
> * Commented out forms are missing ⚠️
> 
> ##### `new`
> ###### commented out `form`
> * [x]  uses `users_path` helper for action
> * [x]  uses `post` for method
> * [x]  uses `"UTF-8"` for ccept-charset
> * [x]  uses `user[<field_name>]` for input's name
> * [x]  uses form_authenticity_token
> 
> ```
> <form action=users_path method='post' accept-charset="UTF-8">
>   <label for='user[username]'>Username</label>
>   <input name='user[username]' type='text'>
>   <input type="hidden" name="authenticity_token" value="<%= form_authenticity_token %>">
>   <input type="submit" value="Submit">
> </form>
> ```
> 
> ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
> 
> ###### commented out `form_tag`
> * [x]  uses `label_tag`, `text_field_tag`, `email_field_tag`, `password_field_tag`
> 
> ```
> <%= form_tag(users_path) do |f|%>
>   <%= label_tag 'username' %>
>   <%= text_field_tag 'username' %>
>   <%= label_tag 'email' %>
>   <%= email_field_tag 'email' %>
>   <%= label_tag 'password' %>
>   <%= password_field_tag 'password' %>
>   <%= submit_tag %>
> <% end %>
> ```
> 
> ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_tag2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
> 
> ###### `form_for`
> * Errors get automatically generated in the view when using form_for
> 
> * [x]  uses `@user` from controller
> 
> ```
> <%= form_for @user do |u|%>
>   <%= u.label :username %>
>   <%= u.text_field :username %>
>   <%= u.label :email %>
>   <%= u.email_field :email %>
>   <%= u.label :password %>
>   <%= u.password_field :password %>
>   <%= u.submit %>
> <% end %>
> ```
> 
> ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_for.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
> 
> ##### `edit`
> ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_edit.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
> 
> ### That's it 😌 When finished, kindly submit a code re-review request 🙏
> Happy coding ✌️
> **_Ahmed Mahfoudh_**

Hello @stratospherique Thank you for your review of our project! We have made the changes. Have a great day.

**Review by nikasulo — APPROVED** [review]
# Great work! 🎖️  
You've a wonderful job handling the requested changes. Please merge and keep moving forward, I wish you the very best and see you at the top 🚀

**Comment by Calvinoea** [comment]
> # Great work! 🎖️
> You've a wonderful job handling the requested changes. Please merge and keep moving forward, I wish you the very best and see you at the top 🚀

Thank you for your review and wishes @Oluwadamilareolusakin ! See you at the top and best wishes 🙏🏿

**Pull request merged**